lib/assert.js

```diff
@@ -89,6 +89,17 @@ function innerFail(obj) {
   throw new AssertionError(obj);
 }
 
+function looseEqual(a, b) {
+  if (a === b) {
+    return true;
+  }
+  if (typeof a === 'object' || typeof b === 'object') {
+    return a === undefined || b === undefined;
```

Comment:
Can you explain why it's desirable to not throw for `assert.equal(undefined, {})` ?

Reply:
Good catch. This was not intentional.

Reply:
I think this should work:
```js
if (a == null && b == null) {
  return a === b;
}
```

Reply:
well, it depends if `assert.equal()` should return true or false for `null, undefined`

Reply:
I updated it and it returns true for `assert.equal(null, undefined)` while not considering any other objects.